README.md

```diff
@@ -1,85 +1,165 @@
+[AWS environment variables](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html).
+
+The Query Service uses [Terraform](https://www.terraform.io/) to manage the deployment process. Before deploying, edit
+the `environment` file to set essential variables to control the deployment:
+
+* `STAGE` - Application deployment stage (such as `dev`, `staging` or `production`)
+* `APP_NAME` - The name that the query service will use to refer to itself
+* `API_DOMAIN_NAME` - The domain name that the query service will register for itself using
+  [Route 53](https://aws.amazon.com/route53/) and [API Gateway](https://aws.amazon.com/api-gateway/). Before continuing,
+  ensure that a certificate for this domain name is available (in VERIFIED state) in the
+  [AWS Certificate Manager](https://console.aws.amazon.com/acm/home). 
+* `API_DNS_ZONE` - The name of a [Route 53](https://aws.amazon.com/route53/) zone that must be present in your AWS
+  account, with a registered domain pointing to it. You can set up both at https://console.aws.amazon.com/route53/.
+
+Run `source environment` to set the results of the above edits in your shell.
+
+In the same shell, run `make install-secrets` to initialize and upload credentials that Query Service
+components need to communicate with each other.
+
+Finally, to deploy the Query Service, run `make deploy` in the same shell.
+
+#### Minor app updates
+
+After deploying, you can update just the Lambda function codebase by running `make update-lambda` (this is faster, but
+no dependencies, routes, or IAM policies will be updated).
+
+If you are using an AWS Assume Role profile, credentials expire in 15 minutes by default. Terraform does not integrate
+with the AWS CLI, and will not automatically refresh these credentials. If your commands fail with an "ExpiredToken"
+error, run `make init-tf` to refresh the credentials. To lengthen the lifetime of your Assume Role token, add a line
+like `duration_seconds = 43200` in your profile in `~/.aws/config`.
+
+#### Database management
+
+Run `python -m dcpquery.db --help` for a list of commands available to maange the database. By default, this script
+connects to the PostgreSQL database running on localhost. To connect to the remote database listed in the service
+configuration, add `--db remote` to the command.
+
+* Running `python -m dcpquery.db connect` will connect to the database using psql.
+
+* Running `python -m dcpquery.db load` (or `make load`) will load all data from DSS. This requires substantial time and
+  resources. To load a test dataset for experimenting, see "Loading Test Data".
+
+#### Connecting directly to the database
+
+The following command can be used 
+python -m dcpquery.db connect --db remote
+
 ### Testing
-```bash
-make install tests -C query/
-```
 
-### Debugging/Troubleshooting
+Run `make test` to run unit tests. Run `make integration_test` to run integration tests, which require the Query Service
```

Comment:
Is this integration test also a TODO? If so, create a ticket for it to make sure we don't forget to do this.

Reply:
Yes, TODO. Added https://github.com/HumanCellAtlas/query-service/issues/92.